conda_build/main_build.py

```diff
@@ -308,6 +309,13 @@ def execute(args, parser):
                 raise RuntimeError("%s must be major.minor, not %s" %
                     (conda_version[lang], version))
 
+    # Using --python, --numpy etc. is equivalent to using CONDA_PY, CONDA_NPY, etc.
+    # Auto-set those env variables
+    for var in conda_version.values():
+        if getattr(config, var):
```

Comment:
That was already the case.  The original behavior was equivalent to the following pseudo-code:

```python
# By default, use os.environ
conda_build.config.Config.CONDA_PY = os.environ['CONDA_PY']

# Override with --python=foo.bar, if available
if args.python:
    conda_build.config.Config.CONDA_PY = args.python
```

I just want `os.environ` to *also* reflect the change.  In the pseudo-code, I'm adding a line to the if-block:

```python
# By default, use os.environ
conda_build.config.Config.CONDA_PY = os.environ['CONDA_PY']

# Override with --python=foo.bar, if available
if args.python:
    conda_build.config.Config.CONDA_PY = args.python
    # Also, update os.environ
    os.environ['CONDA_PY'] = args.python
```